File: backend/calApp/core/src/main/java/com/caliq/core/dto/MealDto.java
package com.caliq.core.dto;
import java.math.BigDecimal;
import java.util.List;
public record MealDto(Dishes dishes) {
    public record Dishes(List<Dish> dish) {
    }
    public record Dish(
            String type,
            String code,
            String name,
            BigDecimal calories,
            BigDecimal fat,
            BigDecimal protein,
            BigDecimal carbohydrates,
            BigDecimal sugars,
            BigDecimal fiber
    ) {
    }
}

File: androidApp/NytriSync/app/src/main/java/com/ga3t/nytrisync/data/model/MealDto.kt
package com.ga3t.nytrisync.data.model

import java.math.BigDecimal

data class MealDto(
    val dishes: Dishes
) {
    data class Dishes(
        val dish: List<Dish>
    )

    data class Dish(
        val type: String,
        val code: String,
        val name: String,
        val calories: BigDecimal,
        val fat: BigDecimal,
        val protein: BigDecimal,
        val carbohydrates: BigDecimal,
        val sugars: BigDecimal,
        val fiber: BigDecimal
    )
}